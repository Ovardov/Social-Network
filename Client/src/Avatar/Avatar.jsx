import React from 'react';
import { Link } from 'react-router-dom';
import styles from './avatar.module.scss';

function Avatar({ username, name, profilePicture, setIsOpen }) {
    return (
        <span className={styles.container} onClick={() => setIsOpen(true)}>
            <Link to={`/profile/${username}`}>
                <img className={styles["avatar-image"]} src={profilePicture} alt={name}/>
            </Link>
        </span>
    )
}

export default Avatar;