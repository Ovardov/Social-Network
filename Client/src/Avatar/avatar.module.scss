.container {
    width: 3em;
    height: 3em;
    position: relative;
    overflow: hidden;
    cursor: pointer;

    &:after {
        display: block;
        content: '';
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        border: 1px solid rgba(0, 0, 0, 0.1);
        border-radius: 50%;
    }

    & .avatar-image {
        width: 100%;
        height: 100%;
        border-radius: 50%;
    }
}