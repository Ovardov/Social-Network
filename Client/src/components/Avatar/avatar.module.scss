.container {
    width: 3em;
    height: 3em;
    position: relative;
    overflow: hidden;
    cursor: pointer;
    display: block;

    & a {
        position: absolute;
        height: 100%;
        width: 100%;

        &:after {
            display: block;
            content: '';
            position: absolute;
            top: 0;
            right: 0;
            bottom: 0;
            left: 0;
            border: 1px solid rgba(0, 0, 0, 0.1);
            border-radius: 50%;
        }
    }

    & .avatar-image {
        width: 100%;
        height: 100%;
        border-radius: 50%;
    }
}