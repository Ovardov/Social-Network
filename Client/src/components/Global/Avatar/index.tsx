// Libraries
import React, { FC as FC_, useMemo } from 'react';
// Utils
import { getTimeDifference } from '../../../utils/date';
import { Sizes } from '../../../utils/enums';
// Models
import User_ from '../../../models/User';
// Images
import defaultProfilePicture from '../../../../public/images/profile-placeholder.png';
// Styles
import styles from './index.module.scss';
import { useHistory } from 'react-router-dom';

interface Props {
  type: 'image' | 'image-with-info',
  size: Sizes,
  createdAt?: Date,
  user: User_,
  stopRedirectToProfile?: boolean
}

const Avatar: FC_<Props> = ({ type, size, user, createdAt, stopRedirectToProfile = false, }) => {
  const { push, } = useHistory();

  const { profilePicture, fullName, username, } = user;
  const timeDifference = useMemo(() => getTimeDifference(createdAt), [createdAt]);


  return (
    <div
      className={`${styles.container} ${type === 'image-with-info' ? styles.center : ''}`}
      onClick={stopRedirectToProfile ? null: () => push(`/profile/${username}`)}
    >
      <img
        className={`${styles.avatar} ${styles[`size-${size}`]}`}
        src={profilePicture?.imageUrl || defaultProfilePicture}
        alt={fullName}
        loading='lazy'
      />

      {type === 'image-with-info' && (
        <div className={styles.author}>
          {fullName && <h2 className={styles.name}>{fullName}</h2>}
          {createdAt && <p className={styles.date}>{timeDifference}</p>}
        </div>
      )}
    </div>
  );
};

export default Avatar;
