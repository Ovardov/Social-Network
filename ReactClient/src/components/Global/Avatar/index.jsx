// Libraries
import React from 'react'
import PropTypes from 'prop-types'
// Images
import defaultProfilePicture from '../../../../public/images/profile-placeholder.png'
// Styles
import styles from './index.module.scss'

const Avatar = ({ size, imageSrc, imageAlt }) => {
  return (
    <img
      className={`${styles.avatar} ${styles[`size-${size}`]}`}
      // If user do not have profile picture we showed default
      src={imageSrc ? imageSrc : defaultProfilePicture}
      alt={imageAlt}
    />
  )
}

Avatar.propTypes = {
  size: PropTypes.oneOf(['sm', 'md', 'lg']).isRequired,
  imageSrc: PropTypes.string,
  imageAlt: PropTypes.string.isRequired,
}

export default Avatar
