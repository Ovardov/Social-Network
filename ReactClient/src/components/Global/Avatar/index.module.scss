.container {

  &.center {
    display: flex;
    align-items: center;
  }

  .avatar {
    object-fit: cover;
    object-position: center;
    border-radius: 50%;
    box-shadow: $avatar-box-shadow;
    display: block;

    &.size-sm {
      width: 30px;
      height: 30px;

      @include bp(phone) {
        width: 24px;
        height: 24px;
      }
    }

    &.size-md {
      width: 3.5em;
      height: 3.5em;
    }

    &.size-lg {
      width: 5em;
      height: 5em;
    }

    &.size-xl {
      width: 7em;
      height: 7em;
    }
  }

  .author {
    margin-left: $sm-spacing;
    
    .name {
      font-size: 1rem;
      margin-bottom: 0.1em;
      padding-bottom: 0;
      font-family: $font-main, sans-serif;
    }

    .date {
      padding-bottom: 0;
      color: var(--theme-text-darker-color);
    }
  }
}