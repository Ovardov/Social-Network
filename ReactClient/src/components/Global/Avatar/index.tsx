// Libraries
import React, { FC } from 'react'
// Images
import defaultProfilePicture from '../../../../public/images/profile-placeholder.png'
// Models
import { Size } from '../../../utils/models'
// Styles
import styles from './index.module.scss'

interface Props {
  size: Size,
  imageSrc: string,
  imageAlt: string
}

const Avatar: FC<Props> = ({ size, imageSrc, imageAlt }) => {
  return (
    <img
      className={`${styles.avatar} ${styles[`size-${size}`]}`}
      // If user do not have profile picture we showed default
      src={imageSrc ? imageSrc : defaultProfilePicture}
      alt={imageAlt}
    />
  )
}

export default Avatar
