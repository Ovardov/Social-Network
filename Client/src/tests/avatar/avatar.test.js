import React from 'react'
import renderer from 'react-test-renderer';
import Avatar from '../../components/Avatar/Avatar';

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('Avatar component', () => {
    test('should render correcly with given image, name and username', () => {
        const component = renderer.create(
            <Avatar username="Ovardov" name="Aleksandar Ovardov" profilePicture="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"/>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
})
