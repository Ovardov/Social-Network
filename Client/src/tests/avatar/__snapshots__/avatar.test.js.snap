// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render correcly with given image, name and username 1`] = `
<span
  className="container"
  onClick={[Function]}
>
  <Link
    to="/profile/Ovardov"
  >
    <img
      alt="Aleksandar Ovardov"
      className="avatar-image"
      src="https://res.cloudinary.com/dxxq5xtsy/image/upload/v1576368427/cxfz26hn4gxeg36cv56q.jpg"
    />
  </Link>
</span>
`;
